### PR TITLE
test: add comprehensive test coverage for components and pages

### PR DESCRIPTION
**Pull request by SnareHanger**
Add tests for previously untested components and utilities:

- utils/useImageFilename.hooks.ts (5 tests)
  - Placeholder replacement with various image numbers
  - Edge cases for filenames without placeholders

- src/components/projectImages.tsx (7 tests)
  - Correct image count rendering
  - Thumbnail path generation
  - Image title attributes
  - Lightbox integration
  - Click handler functionality

- src/pages/index.tsx (10 tests)
  - Layout integration
  - Project listing from JSON
  - Link generation and styling
  - Head component with meta tags and favicons

- src/pages/404.tsx (4 tests)
  - Layout integration
  - Error message display
  - Heading hierarchy

Total test coverage increased from 5 to 29 tests.

### CODE REVIEW

File: src/components/projectImages.test.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    expect(images[2]).toHaveAttribute("title", "test3.jpg");
+  });
+
+  test("renders lightbox component", () => {
+    const { container } = render(
+      <ProjectImages images={mockImages} dir={mockDir} />
+    );
+
+    // The mocked lightbox renders a div
+    expect(container.querySelector("div")).toBeInTheDocument();
```
Comment (SnareHanger) [inline src/components/projectImages.test.tsx]: Should the selector here be a little more specific?
